markup/highlight/highlight.go

```diff
@@ -157,17 +160,32 @@ func (h chromaHighlighter) GetIdentity() identity.Identity {
 }
 
 type HightlightResult struct {
-	innerLow    int
-	innerHigh   int
 	highlighted template.HTML
+
+	inline bool
+
+	innerLow  int
+	innerHigh int
+
+	innerInit sync.Once
+	inner     template.HTML
 }
 
-func (h HightlightResult) Wrapped() template.HTML {
+func (h *HightlightResult) Wrapped() template.HTML {
 	return h.highlighted
 }
 
-func (h HightlightResult) Inner() template.HTML {
-	return h.highlighted[h.innerLow:h.innerHigh]
+func (h *HightlightResult) Inner() template.HTML {
+	h.innerInit.Do(func() {
+		inner := string(h.highlighted[h.innerLow:h.innerHigh])
+		if h.inline {
+			inner = strings.Replace(inner, `<span class="line">`, `<span class="chroma inline">`, 1)
+			inner = strings.Replace(inner, "display:flex;", "", -1)
```

Comment:
@bep am I missing something or would this unconditionally replace the string `display:flex;` anywhere in the user provided code snippet? For example what happens if I do this?

```
{{< highlight css "hl_inline=true" >}}.line { display:flex; }{{< /highlight >}}
```

Reply:
See https://github.com/gohugoio/hugo/pull/9638#issuecomment-1063133736 😄

Reply:
Yea, I will take another look at this day, I had a massive day of "doing everything on my list" yesterday, and this came in late ... I'm sure Chroma has a better API for this.

Reply:
@kaushalmodi ah, thankssorry I missed your comment I was just looking at the diffs.